gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -100,125 +100,538 @@ var fluid = fluid || require("infusion");
                 func: "{dataStore}.revokeClientCredentialsToken"
                     // clientCredentialsTokenId
             }
+        },
+        events: {
+            onGrantAuthorizationCode: null,
+            onAddAuthorization: null,
+            onGetSelectedPreferences: null,
+            onSetSelectedPreferences: null,
+            onGetUnauthorizedClients: null
+        },
+        listeners: {
+            onGrantAuthorizationCode: [{
+                listener: "gpii.oauth2.authorizationService.checkAuthDecision",
+                args: ["{dataStore}", "{codeGenerator}", "{arguments}.0"],
+                namespace: "checkAuthDecision",
+                priority: "first"
```

Comment:
"first" is unnecessary and will interfere with the use of further authors to take advantage of this "escape hatch"